resources/kernel/make_busybox.sh

```diff
@@ -0,0 +1,103 @@
+#!/bin/bash
+
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script illustrates the build steps for `vmlinux-hello-busybox`.
+
+set -e
+
+WORKDIR="/tmp/vmlinux_busybox"
+SOURCE=$(readlink -f "$0")
+TEST_RESOURCE_DIR="$(dirname "$SOURCE")"
+
+# Optionally build a guest that exits.
+[ "$1" == "-halt" ] && HALT=1
+
+KERNEL="linux-4.14.176"
+KERNEL_ARCHIVE="$KERNEL.tar.xz"
+KERNEL_URL="https://cdn.kernel.org/pub/linux/kernel/v4.x/$KERNEL_ARCHIVE"
+
+BUSYBOX="busybox-1.26.0"
+BUSYBOX_ARCHIVE="$BUSYBOX.tar.bz2"
+BUSYBOX_URL="https://busybox.net/downloads/$BUSYBOX_ARCHIVE"
+
+rm -rf "$WORKDIR" && mkdir "$WORKDIR" && cd "$WORKDIR"
+
+# Download kernel sources.
+echo "Downloading kernel..."
+curl "$KERNEL_URL" > "$KERNEL_ARCHIVE"
+echo "Extracting kernel sources..."
+tar xf "$KERNEL_ARCHIVE"
+pushd "$KERNEL" &>/dev/null
+
+# Copy base kernel config.
+cp "$TEST_RESOURCE_DIR/microvm-kernel-initramfs-hello-x86_64.config" .config
+
+# Prepare busybox.
+echo "Downloading busybox..."
+mkdir -p busybox_rootfs
+curl "$BUSYBOX_URL" > "$BUSYBOX_ARCHIVE"
+echo "Extracting busybox..."
+tar xf "$BUSYBOX_ARCHIVE"
+pushd "$BUSYBOX" &>/dev/null
+# Build statically linked busybox.
+cp "$TEST_RESOURCE_DIR/busybox_static_config" .config
+make -j
```

Comment:
This command is freezing my laptop really bad. Same for the other `make -j` :laughing: 

Any chance we can make this configurable instead? I am not sure it is a good idea even for the CI to run with `-j` because we might have multiple agents on the same host and they'll have no available resources to run their jobs.

Reply:
Done. You can now optionally pass `-j <as many as you want>`. The script got ugly, so I refactored it a bit. PTAL

Reply:
Tests take forever now, though. WDYT would be a sweet spot for num cpus to assign?